src/lib/mdns/Discovery_ImplPlatform.cpp

```diff
@@ -437,6 +439,25 @@ CHIP_ERROR DiscoveryImplPlatform::ResolveNodeId(const PeerId & peerId, Inet::IPA
 {
     ReturnErrorOnFailure(Init());
 
+    Inet::IPaAddress addr;
```

Comment:
Could you guard this (and remaining) code with `#if CHIP_CONFIG_MDNS_CACHE_SIZE > 0`? That way all platforms will have a way to bypass the cache in case their DNS-SD implementations already take care of the caching.

Reply:
That's a reasonable idea.   There was talk in the past about  disabling this feature -- this is a elegant way to disable the feature.